Odata-docs/odatacli/getting-started.md

```diff
@@ -0,0 +1,78 @@
+---
+title: "OData Cli"
+description: "How to generate code using the odata-cli dotnet tool"
+
+author: ElizabethOkerio
+ms.author: ElizabethOkerio
+ms.date: 3/30/2022
+ms.topic: article
+ 
+---
+# OData CLI
+
+**Applies To**: [!INCLUDE[appliesto-odataclient](../includes/appliesto-odataclient-v7.md)]
+
+**OData Cli** is a CLI tool that generates strongly-typed C# and Visual Basic client code for a specified OData service. It generates a `DataServiceContext` class to interact with the service and CLR types for each entity type and complex type in the service model.
+
+The extension is available at [Nuget](https://www.nuget.org/packages/Microsoft.OData.Cli/)
+
+## Commands
+
+The **odata-cli** tool has only one command:  
+
+ - **generate** - This is the command that generates strongly typed C# and Visual Basic client code for a specified OData service.  
+
+## Options 
+
+The following is a list of options that can be passed to the **generate** command described above: Most of the options are optional except the `--metadata-uri|-m` and   `--outputdir|-o` options which are required options.  
+
+- `--metadata-uri|-m` - The URI of the metadata document. The value must be set to a valid service document URI or a local file path. 
+
+ - `--file-name|-fn` - The name of the generated file. If not provided, then the default name 'Reference.cs/.vb' is used. 
+
+ - `--custom-headers|-h` - Headers that will get sent along with the request when fetching the metadata document from the service. Format: Header1:HeaderValue, Header2:HeaderValue. 
```

Comment:
Are commas not valid inside header values? Is there a chance the header value could contain a comma?

Reply:
I think the are.

Reply:
Because the headers are separated by commas. So if you have a header value that contains a comma, then parsing of these headers will be incorrect. This is not an issue with the docs, but with the command handler. Maybe in a future iteration we could make it possible to add multiple options for headers, e.g.: `-h Header1:Value1 -h "Accept-Encoding: gzip, deflate" -h...`